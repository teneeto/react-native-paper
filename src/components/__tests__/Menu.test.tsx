import * as React from 'react';
import { Animated, StyleSheet, View } from 'react-native';

import { render, waitFor } from '@testing-library/react-native';
import renderer from 'react-test-renderer';

import Button from '../Button/Button';
import Menu from '../Menu/Menu';
import Portal from '../Portal/Portal';

const styles = StyleSheet.create({
  contentStyle: {
    borderTopLeftRadius: 0,
    borderTopRightRadius: 0,
  },
});

it('renders visible menu', () => {
  const tree = renderer
    .create(
      <Portal.Host>
        <Menu
          visible
          onDismiss={jest.fn()}
          anchor={<Button mode="outlined">Open menu</Button>}
        >
          <Menu.Item onPress={jest.fn()} title="Undo" />
          <Menu.Item onPress={jest.fn()} title="Redo" />
        </Menu>
      </Portal.Host>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders not visible menu', () => {
  const tree = renderer
    .create(
      <Portal.Host>
        <Menu
          visible={false}
          onDismiss={jest.fn()}
          anchor={<Button mode="outlined">Open menu</Button>}
        >
          <Menu.Item onPress={jest.fn()} title="Undo" />
          <Menu.Item onPress={jest.fn()} title="Redo" />
        </Menu>
      </Portal.Host>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders menu with content styles', () => {
  const tree = renderer
    .create(
      <Portal.Host>
        <Menu
          visible
          onDismiss={jest.fn()}
          anchor={<Button mode="outlined">Open menu</Button>}
          contentStyle={styles.contentStyle}
        >
          <Menu.Item onPress={jest.fn()} title="Undo" />
          <Menu.Item onPress={jest.fn()} title="Redo" />
        </Menu>
      </Portal.Host>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('uses the default anchorPosition of top', async () => {
  function makeMenu(visible: boolean) {
    return (
      <Portal.Host>
        <Menu
          visible={visible}
          onDismiss={jest.fn()}
          anchor={
            <Button mode="outlined" testID="anchor">
              Open menu
            </Button>
          }
          contentStyle={styles.contentStyle}
        >
          <Menu.Item onPress={jest.fn()} title="Undo" />
          <Menu.Item onPress={jest.fn()} title="Redo" />
        </Menu>
      </Portal.Host>
    );
  }

  const tree = renderer.create(makeMenu(false));

  jest
    .spyOn(View.prototype, 'measureInWindow')
    .mockImplementation((fn) => fn(100, 100, 80, 32));

  // You must update instead of creating directly and using it because
  // componentDidUpdate isn't called by default in jest. Forcing the update
  // than triggers measureInWindow, which is how Menu decides where to show
  // itself.
  tree.update(makeMenu(true));

  // If someone has a better way to verify the location of a child on the
  // screen, please by all means let me know.
  await waitFor(() => {
    const menu = tree.root.findByProps({ accessibilityViewIsModal: true });
    expect(menu).toHaveProperty('_fiber.pendingProps.style', [
      {
        position: 'absolute',
      },
      {
        left: 100,
        top: 100,
      },
      undefined,
    ]);
  });
});

it('respects anchorPosition bottom', async () => {
  function makeMenu(visible: boolean) {
    return (
      <Portal.Host>
        <Menu
          visible={visible}
          onDismiss={jest.fn()}
          anchor={
            <Button mode="outlined" testID="anchor">
              Open menu
            </Button>
          }
          anchorPosition="bottom"
          contentStyle={styles.contentStyle}
        >
          <Menu.Item onPress={jest.fn()} title="Undo" />
          <Menu.Item onPress={jest.fn()} title="Redo" />
        </Menu>
      </Portal.Host>
    );
  }

  const tree = renderer.create(makeMenu(false));

  jest
    .spyOn(View.prototype, 'measureInWindow')
    .mockImplementation((fn) => fn(100, 100, 80, 32));

  // See note from previous test about this update.
  tree.update(makeMenu(true));

  await waitFor(() => {
    const menu = tree.root.findByProps({ accessibilityViewIsModal: true });
    expect(menu).toHaveProperty('_fiber.pendingProps.style', [
      {
        position: 'absolute',
      },
      {
        left: 100,
        top: 132,
      },
      undefined,
    ]);
  });
});

it('animated value changes correctly', () => {
  const value = new Animated.Value(1);
  const { getByTestId } = render(
    <Portal.Host>
      <Menu
        visible
        onDismiss={jest.fn()}
        anchor={<Button mode="outlined">Open menu</Button>}
        testID="menu"
        contentStyle={[{ transform: [{ scale: value }] }]}
      >
        <Menu.Item onPress={jest.fn()} title="Test" />
      </Menu>
    </Portal.Host>
  );
  expect(getByTestId('menu-surface-outer-layer')).toHaveStyle({
    transform: [{ scale: 1 }],
  });

  Animated.timing(value, {
    toValue: 1.5,
    useNativeDriver: false,
    duration: 200,
  }).start();

  jest.advanceTimersByTime(200);

  expect(getByTestId('menu-surface-outer-layer')).toHaveStyle({
    transform: [{ scale: 1.5 }],
  });
});
