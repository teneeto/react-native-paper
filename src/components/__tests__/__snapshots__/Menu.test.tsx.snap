// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders menu with content styles 1`] = `
Array [
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      collapsable={false}
    >
      <View
        collapsable={false}
        style={
          Object {
            "alignSelf": undefined,
            "backgroundColor": "transparent",
            "borderRadius": 20,
            "bottom": undefined,
            "end": undefined,
            "flex": undefined,
            "height": undefined,
            "left": undefined,
            "opacity": undefined,
            "position": undefined,
            "right": undefined,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
            "start": undefined,
            "top": undefined,
            "transform": undefined,
            "width": undefined,
          }
        }
        testID="button-container-outer-layer"
      >
        <View
          collapsable={false}
          style={
            Object {
              "backgroundColor": "transparent",
              "borderColor": "rgba(121, 116, 126, 1)",
              "borderRadius": 20,
              "borderStyle": "solid",
              "borderWidth": 1,
              "flex": undefined,
              "minWidth": 64,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
            }
          }
          testID="button-container"
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "disabled": true,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                false,
                Object {
                  "borderRadius": 20,
                },
              ]
            }
            testID="button"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              <Text
                numberOfLines={1}
                selectable={false}
                style={
                  Array [
                    Object {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "color": "rgba(28, 27, 31, 1)",
                      "writingDirection": "ltr",
                    },
                    Array [
                      Object {
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      false,
                      Object {
                        "marginHorizontal": 24,
                        "marginVertical": 10,
                      },
                      undefined,
                      false,
                      Object {
                        "color": "rgba(103, 80, 164, 1)",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 20,
                      },
                      undefined,
                    ],
                  ]
                }
                testID="button-text"
              >
                Open menu
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>,
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      accessibilityLabel="Close menu"
      accessibilityRole="button"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
    <View
      accessibilityViewIsModal={true}
      collapsable={false}
      onAccessibilityEscape={[MockFunction]}
      pointerEvents="box-none"
      style={
        Array [
          Object {
            "position": "absolute",
          },
          Object {
            "left": 8,
            "top": 8,
          },
          undefined,
        ]
      }
    >
      <View
        collapsable={false}
        style={
          Object {
            "transform": Array [
              Object {
                "translateX": -0,
              },
              Object {
                "translateY": -0,
              },
            ],
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "alignSelf": undefined,
              "backgroundColor": "rgb(243, 237, 246)",
              "borderRadius": 4,
              "borderTopLeftRadius": 0,
              "borderTopRightRadius": 0,
              "bottom": undefined,
              "end": undefined,
              "flex": undefined,
              "height": undefined,
              "left": undefined,
              "opacity": 0,
              "position": undefined,
              "right": undefined,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.15,
              "shadowRadius": 6,
              "start": undefined,
              "top": undefined,
              "transform": Array [
                Object {
                  "scaleX": 0,
                },
                Object {
                  "scaleY": 0,
                },
              ],
              "width": undefined,
            }
          }
          testID="menu-surface-outer-layer"
        >
          <View
            collapsable={false}
            style={
              Object {
                "backgroundColor": "rgb(243, 237, 246)",
                "borderRadius": 4,
                "borderTopLeftRadius": 0,
                "borderTopRightRadius": 0,
                "flex": undefined,
                "paddingVertical": 8,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 1,
                  "width": 0,
                },
                "shadowOpacity": 0.3,
                "shadowRadius": 2,
              }
            }
            testID="menu-surface"
          >
            <View
              accessibilityRole="menuitem"
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  false,
                  false,
                  Array [
                    Object {
                      "height": 48,
                      "justifyContent": "center",
                      "maxWidth": 280,
                      "minWidth": 112,
                    },
                    Object {
                      "paddingHorizontal": 12,
                    },
                    undefined,
                    undefined,
                  ],
                ]
              }
              testID="menu-item"
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Array [
                      false,
                      Object {
                        "justifyContent": "center",
                      },
                      Object {
                        "maxWidth": 268,
                        "minWidth": 100,
                      },
                      Object {
                        "marginLeft": 4,
                      },
                      undefined,
                    ]
                  }
                >
                  <Text
                    maxFontSizeMultiplier={1.5}
                    numberOfLines={1}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "fontFamily": "System",
                          "fontSize": 16,
                          "fontWeight": "400",
                          "letterSpacing": 0.15,
                          "lineHeight": 24,
                        },
                        Object {
                          "textAlign": "left",
                        },
                        Object {
                          "color": "rgba(28, 27, 31, 1)",
                          "writingDirection": "ltr",
                        },
                        Array [
                          false,
                          Object {
                            "color": "rgba(28, 27, 31, 1)",
                            "fontFamily": "System",
                            "fontSize": 16,
                            "fontWeight": "400",
                            "letterSpacing": 0.15,
                            "lineHeight": 24,
                          },
                          undefined,
                        ],
                      ]
                    }
                    testID="menu-item-title"
                  >
                    Undo
                  </Text>
                </View>
              </View>
            </View>
            <View
              accessibilityRole="menuitem"
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  false,
                  false,
                  Array [
                    Object {
                      "height": 48,
                      "justifyContent": "center",
                      "maxWidth": 280,
                      "minWidth": 112,
                    },
                    Object {
                      "paddingHorizontal": 12,
                    },
                    undefined,
                    undefined,
                  ],
                ]
              }
              testID="menu-item"
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Array [
                      false,
                      Object {
                        "justifyContent": "center",
                      },
                      Object {
                        "maxWidth": 268,
                        "minWidth": 100,
                      },
                      Object {
                        "marginLeft": 4,
                      },
                      undefined,
                    ]
                  }
                >
                  <Text
                    maxFontSizeMultiplier={1.5}
                    numberOfLines={1}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "fontFamily": "System",
                          "fontSize": 16,
                          "fontWeight": "400",
                          "letterSpacing": 0.15,
                          "lineHeight": 24,
                        },
                        Object {
                          "textAlign": "left",
                        },
                        Object {
                          "color": "rgba(28, 27, 31, 1)",
                          "writingDirection": "ltr",
                        },
                        Array [
                          false,
                          Object {
                            "color": "rgba(28, 27, 31, 1)",
                            "fontFamily": "System",
                            "fontSize": 16,
                            "fontWeight": "400",
                            "letterSpacing": 0.15,
                            "lineHeight": 24,
                          },
                          undefined,
                        ],
                      ]
                    }
                    testID="menu-item-title"
                  >
                    Redo
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>,
]
`;

exports[`renders not visible menu 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={false}
  >
    <View
      collapsable={false}
      style={
        Object {
          "alignSelf": undefined,
          "backgroundColor": "transparent",
          "borderRadius": 20,
          "bottom": undefined,
          "end": undefined,
          "flex": undefined,
          "height": undefined,
          "left": undefined,
          "opacity": undefined,
          "position": undefined,
          "right": undefined,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
          "start": undefined,
          "top": undefined,
          "transform": undefined,
          "width": undefined,
        }
      }
      testID="button-container-outer-layer"
    >
      <View
        collapsable={false}
        style={
          Object {
            "backgroundColor": "transparent",
            "borderColor": "rgba(121, 116, 126, 1)",
            "borderRadius": 20,
            "borderStyle": "solid",
            "borderWidth": 1,
            "flex": undefined,
            "minWidth": 64,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="button-container"
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": true,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              false,
              Object {
                "borderRadius": 20,
              },
            ]
          }
          testID="button"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                Array [
                  Object {
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "letterSpacing": 0.1,
                    "lineHeight": 20,
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "color": "rgba(28, 27, 31, 1)",
                    "writingDirection": "ltr",
                  },
                  Array [
                    Object {
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    false,
                    Object {
                      "marginHorizontal": 24,
                      "marginVertical": 10,
                    },
                    undefined,
                    false,
                    Object {
                      "color": "rgba(103, 80, 164, 1)",
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    undefined,
                  ],
                ]
              }
              testID="button-text"
            >
              Open menu
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders visible menu 1`] = `
Array [
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      collapsable={false}
    >
      <View
        collapsable={false}
        style={
          Object {
            "alignSelf": undefined,
            "backgroundColor": "transparent",
            "borderRadius": 20,
            "bottom": undefined,
            "end": undefined,
            "flex": undefined,
            "height": undefined,
            "left": undefined,
            "opacity": undefined,
            "position": undefined,
            "right": undefined,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
            "start": undefined,
            "top": undefined,
            "transform": undefined,
            "width": undefined,
          }
        }
        testID="button-container-outer-layer"
      >
        <View
          collapsable={false}
          style={
            Object {
              "backgroundColor": "transparent",
              "borderColor": "rgba(121, 116, 126, 1)",
              "borderRadius": 20,
              "borderStyle": "solid",
              "borderWidth": 1,
              "flex": undefined,
              "minWidth": 64,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
            }
          }
          testID="button-container"
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "disabled": true,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                false,
                Object {
                  "borderRadius": 20,
                },
              ]
            }
            testID="button"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              <Text
                numberOfLines={1}
                selectable={false}
                style={
                  Array [
                    Object {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "color": "rgba(28, 27, 31, 1)",
                      "writingDirection": "ltr",
                    },
                    Array [
                      Object {
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      false,
                      Object {
                        "marginHorizontal": 24,
                        "marginVertical": 10,
                      },
                      undefined,
                      false,
                      Object {
                        "color": "rgba(103, 80, 164, 1)",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 20,
                      },
                      undefined,
                    ],
                  ]
                }
                testID="button-text"
              >
                Open menu
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>,
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      accessibilityLabel="Close menu"
      accessibilityRole="button"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
    <View
      accessibilityViewIsModal={true}
      collapsable={false}
      onAccessibilityEscape={[MockFunction]}
      pointerEvents="box-none"
      style={
        Array [
          Object {
            "position": "absolute",
          },
          Object {
            "left": 8,
            "top": 8,
          },
          undefined,
        ]
      }
    >
      <View
        collapsable={false}
        style={
          Object {
            "transform": Array [
              Object {
                "translateX": -0,
              },
              Object {
                "translateY": -0,
              },
            ],
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "alignSelf": undefined,
              "backgroundColor": "rgb(243, 237, 246)",
              "borderRadius": 4,
              "bottom": undefined,
              "end": undefined,
              "flex": undefined,
              "height": undefined,
              "left": undefined,
              "opacity": 0,
              "position": undefined,
              "right": undefined,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.15,
              "shadowRadius": 6,
              "start": undefined,
              "top": undefined,
              "transform": Array [
                Object {
                  "scaleX": 0,
                },
                Object {
                  "scaleY": 0,
                },
              ],
              "width": undefined,
            }
          }
          testID="menu-surface-outer-layer"
        >
          <View
            collapsable={false}
            style={
              Object {
                "backgroundColor": "rgb(243, 237, 246)",
                "borderRadius": 4,
                "flex": undefined,
                "paddingVertical": 8,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 1,
                  "width": 0,
                },
                "shadowOpacity": 0.3,
                "shadowRadius": 2,
              }
            }
            testID="menu-surface"
          >
            <View
              accessibilityRole="menuitem"
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  false,
                  false,
                  Array [
                    Object {
                      "height": 48,
                      "justifyContent": "center",
                      "maxWidth": 280,
                      "minWidth": 112,
                    },
                    Object {
                      "paddingHorizontal": 12,
                    },
                    undefined,
                    undefined,
                  ],
                ]
              }
              testID="menu-item"
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Array [
                      false,
                      Object {
                        "justifyContent": "center",
                      },
                      Object {
                        "maxWidth": 268,
                        "minWidth": 100,
                      },
                      Object {
                        "marginLeft": 4,
                      },
                      undefined,
                    ]
                  }
                >
                  <Text
                    maxFontSizeMultiplier={1.5}
                    numberOfLines={1}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "fontFamily": "System",
                          "fontSize": 16,
                          "fontWeight": "400",
                          "letterSpacing": 0.15,
                          "lineHeight": 24,
                        },
                        Object {
                          "textAlign": "left",
                        },
                        Object {
                          "color": "rgba(28, 27, 31, 1)",
                          "writingDirection": "ltr",
                        },
                        Array [
                          false,
                          Object {
                            "color": "rgba(28, 27, 31, 1)",
                            "fontFamily": "System",
                            "fontSize": 16,
                            "fontWeight": "400",
                            "letterSpacing": 0.15,
                            "lineHeight": 24,
                          },
                          undefined,
                        ],
                      ]
                    }
                    testID="menu-item-title"
                  >
                    Undo
                  </Text>
                </View>
              </View>
            </View>
            <View
              accessibilityRole="menuitem"
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  false,
                  false,
                  Array [
                    Object {
                      "height": 48,
                      "justifyContent": "center",
                      "maxWidth": 280,
                      "minWidth": 112,
                    },
                    Object {
                      "paddingHorizontal": 12,
                    },
                    undefined,
                    undefined,
                  ],
                ]
              }
              testID="menu-item"
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Array [
                      false,
                      Object {
                        "justifyContent": "center",
                      },
                      Object {
                        "maxWidth": 268,
                        "minWidth": 100,
                      },
                      Object {
                        "marginLeft": 4,
                      },
                      undefined,
                    ]
                  }
                >
                  <Text
                    maxFontSizeMultiplier={1.5}
                    numberOfLines={1}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "fontFamily": "System",
                          "fontSize": 16,
                          "fontWeight": "400",
                          "letterSpacing": 0.15,
                          "lineHeight": 24,
                        },
                        Object {
                          "textAlign": "left",
                        },
                        Object {
                          "color": "rgba(28, 27, 31, 1)",
                          "writingDirection": "ltr",
                        },
                        Array [
                          false,
                          Object {
                            "color": "rgba(28, 27, 31, 1)",
                            "fontFamily": "System",
                            "fontSize": 16,
                            "fontWeight": "400",
                            "letterSpacing": 0.15,
                            "lineHeight": 24,
                          },
                          undefined,
                        ],
                      ]
                    }
                    testID="menu-item-title"
                  >
                    Redo
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>,
]
`;
